### PR TITLE
James 1741

### PR DESCRIPTION
**Pull request by chibenwa**
- Classes like CassandraMailboxSessionFactory shuld be bound in scope singleton
 - Singleton statements should be located in Guice modules
 - Uniform uses between .in(Scopes.SINGLETON) rather than .in(Singleton.class) in Guice modules

### CODE REVIEW

**Comment by chibenwa** [comment]
Diagnoses : 

## @Singleton annotations

Quota (not yet integrated to Guice deployement) relies a lot on Singleton

List of annotated JMAP class :

* JMAPServer
* SignedContinuationTokenManager
* JamesSignatureHandler
* AccessTokenManagerImpl

* CassandraMailboxManager

InMemoryMailRepositoryStore
* FileConfigurationProvider

* EmbeddedActiveMQ
* MemoryMailQueueFactory

* DNSJavaService

* FileSystemImpl

## Not bind in singleton today

*CassandraMailboxSessionMapperFactory
*CassandraModSeqProvider
*CassandraUidProvider
*CassandraSubscriptionManager
*(NoMailboxPathLocker)
*UserRepositoryAuthenticator

*ElasticSearchListeningMessageSearchIndex
*TikaTextExtractor

* SieveParser

**Comment by rouazana** [comment]
:+1:

File: server/container/guice/cassandra-guice/src/main/java/org/apache/james/modules/mailbox/CassandraMailboxModule.java [hunk 1]
```diff
@@ -52,17 +52,25 @@
...
+        bind(NoMailboxPathLocker.class).in(Scopes.SINGLETON);
+        bind(CassandraSubscriptionManager.class).in(Scopes.SINGLETON);
+        bind(CassandraModSeqProvider.class).in(Scopes.SINGLETON);
+        bind(CassandraUidProvider.class).in(Scopes.SINGLETON);
+        bind(UserRepositoryAuthenticator.class).in(Scopes.SINGLETON);
+
         bind(MessageMapperFactory.class).to(CassandraMailboxSessionMapperFactory.class);
         bind(MailboxMapperFactory.class).to(CassandraMailboxSessionMapperFactory.class);
         bind(MailboxSessionMapperFactory.class).to(CassandraMailboxSessionMapperFactory.class);
+        bind(MailboxSessionMapperFactory.class).to(CassandraMailboxSessionMapperFactory.class);
```
Comment (aduprat) [inline server/container/guice/cassandra-guice/src/main/java/org/apache/james/modules/mailbox/CassandraMailboxModule.java]: Why do you duplicate this line?

**Comment by chibenwa** [comment]
test this please

**Comment by rouazana** [comment]
:+1:

**Comment by aduprat** [comment]
:+1:

**Comment by mbaechler** [comment]
👍

**Pull request closed without merging**